modules/core/src/main/java/org/apache/ignite/internal/cache/query/index/IndexQueryProcessor.java

```diff
@@ -125,6 +131,38 @@ public <K, V> IndexQueryResult<K, V> queryLocal(
 
                 IgniteBiTuple<K, V> row = currVal;
 
+                if (cctx.events().isRecordable(EVT_CACHE_QUERY_OBJECT_READ) &&
+                    cctx.gridEvents().hasListener(EVT_CACHE_QUERY_OBJECT_READ)) {
```

Comment:
Actually, there is no need in checking listeners. In some places this is checked, in some - not. Let's skip this check.

Reply:
Check for available listeners removed.